lib/commands/base-punishment-helper.js

```diff
@@ -6,14 +6,14 @@ function basePunishmentHelper(input, defaultDuration, overRideDuration) {
   const matched = /(\d+[hmds])?\s?(\w+)/.exec(input);
   const userToPunish = _.get(matched, 2, '').toLowerCase();
   let parsedDuration = 0;
-  if (parsedDuration !== false) {
-    parsedDuration = _.get(matched, 1, '').toLowerCase();
-    switch (parsedDuration) {
+  if (!overRideDuration) {
```

Comment:
Nice catch, how about !== undefined though. A value of 0 here triggers the wrong logic.

Reply:
Good point